socks5.py

```diff
@@ -325,8 +324,12 @@ def handle_connect(self, address, port):
             ipv4 = address.get('ipv4')
             ipv6 = address.get('ipv6')
         else:
-            ipv4 = address
-            ipv6 = None
+            if ':' in address:
+                ipv6 = address
+                ipv4 = None
+            else:
+                ipv4 = address
+                ipv6 = None
 
         if (ipv6 is not None and "ipv6" in CONNECT_HOST and CONNECT_HOST["ipv6"]) or (ipv4 is not None and "ipv4" in CONNECT_HOST and CONNECT_HOST["ipv4"]):
```

Comment:
I think the `or` in the middle needs to be an `and`, since this is the branch that leads to happy eyeballs?